pkg/utils/client/metadata/metadata.go

```diff
@@ -0,0 +1,257 @@
+/*
+Copyright 2024 The CloudPilot AI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metadata
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"reflect"
+	"strings"
+	"time"
+)
+
+const (
+	Endpoint    = "http://100.100.100.200"
+	hostname    = "hostname"
+	instanceID  = "instance-id"
+	regionID    = "region-id"
+	ramSecurity = "ram/security-credentials"
+)
+
+// MetaData wrap http client
+type MetaData struct {
+	// mock for unit test.
+	mock   requestMock
+	client *http.Client
+}
+
+// NewMetaData returns MetaData
+func NewMetaData(client *http.Client) *MetaData {
+	if client == nil {
+		client = &http.Client{}
+	}
+	return &MetaData{
+		client: client,
+	}
+}
+
+// NewMockMetaData returns mock MetaData
+func NewMockMetaData(client *http.Client, sendRequest requestMock) *MetaData {
+	if client == nil {
+		client = &http.Client{}
+	}
+	return &MetaData{
+		client: client,
+		mock:   sendRequest,
+	}
+}
+
+// New returns MetaDataRequest
+func (m *MetaData) New() *MetaDataRequest {
+	return &MetaDataRequest{
+		client:      m.client,
+		sendRequest: m.mock,
+	}
+}
+
+// HostName returns host name
+func (m *MetaData) HostName() (string, error) {
```

Comment:
Maybe we should only keep what we need and delete the unused function?

Reply:
It's really redundant. I will clean.